src/main/java/pl/put/poznan/transformer/transformation/EmojiTextTransformation.java

```diff
@@ -0,0 +1,65 @@
+package pl.put.poznan.transformer.transformation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Map.entry;
+
+/**
+ * Class proving transformation of selected symbols into latex format
+ * @author Nina Zukowska
+ * @version 2.2
+ */
+
+public class EmojiTextTransformation extends TextTransformation {
+    public static final String NAME = "emoji";
+
+    Map<String, String> emojiMap = Map.ofEntries(
+            entry("happy", ":)"),
+            entry("sad", ":("),
+            entry("laugh", ":D"),
+            entry("meh", ":|"),
+            entry("boss", "B)"),
+            entry("cat", ":3"),
+            entry("wink", ";‑)"),
+            entry("elvis", "5:‑)"),
+            entry("heart", "<3"),
+            entry("surprised", "O_O"),
+            entry("troubled", "(>_<)"),
+            entry("tired", "(=_=)"),
+            entry("headphones", "((d[-_-]b))"),
+            entry("suspicious", "(p_-)")
+    );
+
+
+    public EmojiTextTransformation() {
+        super();
+    }
+
+    public EmojiTextTransformation(final TextTransformation previousTransformation) {
+        super(previousTransformation);
+    }
+
+    private String emoji(final String text) {
+        String[] texts = {text};
```

Comment:
this logic seems overcomplicated to me 😅, the array is not really used, maybe simply:
```
        for(HashMap.Entry<String, String> emoji : emojiMap.entrySet())
        {
            text = text.replace(emoji.getKey(), emoji.getValue());
        }
```??